src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  name = '';
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore) {
+      const indexOfHerbivore = Animal.alive.findIndex((animal) => {
+        return animal === herbivore && animal.hidden === false;
+      });
+
+      if (indexOfHerbivore !== -1) {
+        Animal.alive[indexOfHerbivore].health -= 50;
```

Comment:
```suggestion
      herbivore.health -= 50;
```
And 50 is a magic value